src/ast/semantic_analyser.cpp

```diff
@@ -433,6 +433,22 @@ void SemanticAnalyser::visit(Call &call)
       }
     }
   }
+  else if (call.func == "buf") {
+    if (check_nargs(call, 2)) {
+      auto &arg = *call.vargs->at(0);
+      if (!(arg.type.type == Type::integer || arg.type.type == Type::string)) {
+        error(call.func + "() expects an integer or string argument",
+              call.loc);
+      }
+      call.type = SizedType(Type::buffer, bpftrace_.strlen_);
```

Comment:
If a user specifies the length they want, do we still need to use `strlen_` by default. It means that every use of `buf` requires 65  bytes, that can blow the 512 byte stack quickly